### PR TITLE
Feature/um 29 nav buttons

### PR DESCRIPTION
**Pull request by Havana82**
## Description

This PR redesigns the navbar to fit  figma design. It also adds the required hover to the buttons. active button issue is now resolved!. note:This branch has been merged to the responsive design branch.

## Acceptance Criteria
Default Button State:

- [x]  When not selected, navigation buttons ('About Us' & 'Resources') should display a solid red background by default.

Selected Button State:

- [x] The button representing the currently active/selected page should be filled with a gradient, as depicted in the provided button image.

Hover State:

- [x] Upon hover, any unselected button should transition to a different shade to indicate interactivity without affecting the selected button's gradient fill.

Persistent Selection Indication:

- [x] Once a user navigates to a page, the corresponding button must retain the gradient fill throughout the session or until a different navigation button is selected.

Transition Animation:

- [x] Implement a smooth transition effect when the button state changes from unselected to selected and vice versa.

## What type of PR is this? (check all applicable)

|     | Type                       |
| --- | -------------------------- |
|     | :sparkles: New feature     |
|  ✓   | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Mobile & Desktop Screenshots

Screenshot 2023-11-19 at 9 19 00 AM

## Testing steps

Switch to this branch with the following command: git switch feature/UM-29-nav-buttons
Run the following scripts: npm run dev to start Vite's dev build 
View the app in browser (the npm run dev script will return the url you can see the app at)
Hover over the navigation buttons to see hover effect, hover over the 'Learn more' buttons on the cards
Navigate to the bio page of one card. Hover over the "return to unmatched voices" button and the "open resource" button
Note Navbar responsiveness is maintained.
On each bio page, read through the copy, check for typos, makes suggestions, and check the links to make sure they work.

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| ✓  | 🙅 no documentation needed    |

  trust me

### CODE REVIEW

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -10,17 +10,13 @@ export default function NavBar() {
...
       <NavLink
       to="/professionals"
-      className="px-4 py-2 bg-base-100 text-primary rounded-full font-bold hover:bg-red-500 hover:text-white"
-      activeClassName="bg-red-500 text-white border-red-500"
-      >
-        Resources
+      className="px-4 py-[8px] rounded-full font-bold bg-red-500 text-white text-[2.1vh] hover:bg-gradient-to-r from-red-500 to-purple-500 ">
+        Unmatched Voices
       </NavLink>
     </div>
```
Comment (shyusu4) [inline src/components/Navbar.jsx:52]:
```
<div className="space-x-6 md:space-x-8 text-sm md:text-base xl:text-lg">
      <NavLink
        exact
        to="/"
        className="text-primary font-bold flex-col justify-start items-center inline-flex relative group"
      >
        <span className="text-center font-bold md:group-hover:opacity-100">
          About Us
        </span>
        <span className="w-full md:h-1 h-[3px] top-5 md:top-full bg-primary rounded-2xl absolute bottom-0 md:opacity-0 transition-opacity duration-300 ease-in-out opacity-100 md:group-hover:opacity-100" />
      </NavLink>
      <NavLink
        exact
        to="/"
        className="text-primary font-bold flex-col justify-start items-center inline-flex relative"
      >
        <span className="md:px-4 md:py-1 rounded-full md:bg-red-500 md:text-white md:hover:bg-gradient-to-r from-red-500 to-purple-500 text-center font-bold">
          Unmatched Voices
        </span>
        <span className="w-full md:h-1 h-[3px] top-5 md:top-full bg-primary rounded-2xl absolute bottom-0 md:opacity-0 transition-opacity duration-300 ease-in-out opacity-100 md:group-hover:opacity-100" />
      </NavLink>
    </div>
```

**Review by shyusu4 — CHANGES_REQUESTED** [review]
Great work! I think this looks great, but I noticed that the navlinks dont have the underlines for smaller screens like they do in the figma design. I modified the code on my end a bit and added it as a comment, so please take a look and see if you think its okay.

File: src/components/Navbar.jsx (hunk 1 above)
Comment (shyusu4) [reply src/components/Navbar.jsx:52]: instead of using `border-b` im using a span to act as the underline so that its easier to style.

Comment (Havana82) [reply src/components/Navbar.jsx:52]: Thank you so much for this Sherry! I totally forgot about the mobile view being different. its updated now but the active class for mobile screen isn't working for now

Comment (Havana82) [reply src/components/Navbar.jsx:52]: Everything is rectified now!

**Review by shyusu4 — APPROVED** [review]

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -1,27 +1,54 @@
 // LIBRARY IMPORTS
-import React from 'react';
+import React, { useState } from 'react';
 import { NavLink } from 'react-router-dom';
 
 // LOCAL IMPORTS
 
 export default function NavBar() {
+  const clicked = {clickedVoiceIcon:false, clickedAboutIcon:false}
```
Comment (avinashi10) [inline src/components/Navbar.jsx:8]: great idea storing these states separately in an object to help with color change! common JS naming convention for booleans usually uses verb 'is' (e.g. `isClicked`)

File: src/components/Navbar.jsx [hunk 3]
```diff
@@ -1,27 +1,54 @@
...
+  
+  const [iconClick, seticonClick] = useState(clicked )
+
+  const voiceIconColor = iconClick.clickedVoiceIcon? 'md:bg-gradient-to-r from-red-500 to-purple-500 ' : 'md:bg-red-500';
+  const aboutIconColor = iconClick.clickedAboutIcon? 'md:border-b-4 border-b-[#0A2E50] ' : '';
+  const voiceIcon = iconClick.clickedVoiceIcon? 'bg-gradient-to-r from-red-500 to-purple-500 text-transparent bg-clip-text' : 'text-primary';
+  const aboutIcon = iconClick.clickedAboutIcon? 'bg-gradient-to-r from-red-500 to-purple-500 bg-clip-text text-transparent' : 'text-primary';
+
+  const changeVoicesColor = ()=>{
+    const isClicked = {...clicked, clickedVoiceIcon:true}
```
Comment (avinashi10) [inline src/components/Navbar.jsx:18]: nice use of the ES6 spread operator here

**Review by avinashi10 — APPROVED** [review]
Looks great!

**Pull request merged**